crates/rust-analyzer/src/reload.rs

```diff
@@ -174,7 +174,11 @@ impl GlobalState {
 
         if let Err(_) = self.fetch_workspace_error() {
             status.health = lsp_ext::Health::Error;
-            message.push_str("Failed to load workspaces.\n\n");
+            message.push_str("Failed to load workspaces.");
+            if self.config.has_linked_projects() {
+                message.push_str(" Note that linked projects have been specified, it looks like this setting has been misconfigured.");
+            }
+            message.push_str("\n\n");
```

Comment:
Sorry, realized that it can be just:
```suggestion
            message.push_str("Failed to load workspaces.");
            if self.config.has_linked_projects() {
                let linked_projects = self.config.linked_projects();
                let (manifest_paths, inline_projects): (Vec<_>, Vec<_>) =
                    linked_projects.iter().partition(|linked_project| {
                        matches!(linked_project, LinkedProject::ProjectManifest(..))
                    });

                message.push_str(
                    "`rust-analyzer.linkedprojects` have been specified, which may cause indexing failures.\n\n ",
                );
                if !manifest_paths.is_empty() {
                    let paths = format!("paths to projects: {:?}", manifest_paths);
                    message.push_str(&paths)
                }

                match inline_projects.len() {
                    0 => (),
                    1 => message.push_str(
                        "The contents of the linked project were redacted due to its length.",
                    ),
                    _ => message.push_str(
                        "The contents of the linked projects were redacted due to its length.",
                    ),
                };
            }
```
feel free to take it or leave it :)